### ISSUE TITLE
COMPOSE_PROJECT_NAME defaults to blank string, not basename of project directory as docs claim

### ISSUE THREAD

**Issue by dossy** [docker/docs#13874]
File: [compose/reference/envvars.md](https://docs.docker.com/compose/reference/envvars/)

The documentation claims:

> Setting this is optional. If you do not set this, the COMPOSE_PROJECT_NAME defaults to the basename of the project directory. See also the -p command-line option.

But, as of docker-compose version 1.29.2, build 5becea4c:

```
$ docker-compose build
WARNING: The COMPOSE_PROJECT_NAME variable is not set. Defaulting to a blank string.
...
```

It would actually be _very_ useful if the behavior of `COMPOSE_PROJECT_NAME` did what the documentation says, but unfortunately it doesn't.

This is a documentation bug (as the documentation is factually incorrect) and possibly a bug in docker-compose, depending on what the code was actually meant to do in this case.

**Comment by Greesb** [comment]
Still have the same issue with docker-compose v2.6.0:
```bash
➜ docker-compose up -d
WARN[0000] The "COMPOSE_PROJECT_NAME" variable is not set. Defaulting to a blank string.
WARN[0000] The "COMPOSE_PROJECT_NAME" variable is not set. Defaulting to a blank string.
```

I am on a MacbookPro M1, if that matters.

**Comment by mfacerias** [comment]
Same problem...

I've tried using tis in .env :
COMPOSE_PROJECT_NAME=`pwd |xargs basename`

I've tried to use $(...) too.

The order is not expended by shell.

**Comment by tthibodeau** [comment]
The `COMPOSE_PROJECT_NAME` variable seems to work as documented when using Windows or WSL2, but running on Linux Docker engine fails with `WARN[0000] The "COMPOSE_PROJECT_NAME" variable is not set. Defaulting to a blank string. `

Temporary fix in linux bash script was to set the variable before the `docker compose` call:
```
PROJECT_NAME="test"

eval "COMPOSE_PROJECT_NAME=${PROJECT_NAME} docker compose --project-name ${PROJECT_NAME} up --detach --build --force-recreate --remove-orphans"
```

Issue on Fedora or Ubuntu Linux
```
❯ docker compose version                                                                  
Docker Compose version v2.6.0
```

**Comment by mfacerias** [comment]
Thanks for comment. But... is it possible to fix this bug ?

Le 2022-08-29 23:49, tthibodeau a écrit :

**Comment by tthibodeau** [comment]
[SOLVED]
I ran into this issue on a different Windows machine as well. I looked at my Docker Compose version and it was v2.6.0.

It seems that this issue is fixed in Docker Compose version v2.7.0.

As Linux seems to currently default to docker-compose v2.6.0, this bug presents itself. Once I updated to docker-compose v2.7.0, it works fine. Had to do a [manual docker-compose install for v2.7.0](https://docs.docker.com/compose/install/compose-plugin/#install-the-plugin-manually)

**Comment by dossy** [comment]
Confirming that this appears to be fixed, testing with Docker Compose v2.12.2:

```
$ docker compose version
Docker Compose version v2.12.2

$ pwd
/Users/dossy/tmp

$ cat <<-EOT >docker-compose.yml
services:
  test:
     image: busybox
    command: echo \${COMPOSE_PROJECT_NAME}
EOT

$ cat docker-compose.yml
services:
  test:
    image: busybox
    command: echo ${COMPOSE_PROJECT_NAME}

$ docker compose run --rm test
tmp
```

Closing as this has been fixed.
